c++/src/kj/compat/http.c++

```diff
@@ -6409,6 +6428,17 @@ private:
       };
     }
 
+    void handleException(kj::Exception&& ex) {
+      // Log the exception...
+      KJ_LOG(ERROR, "Error in HttpClientAdapter connect()", kj::cp(ex));
+      // Reject the status promise if it is still pending...
+      if (fulfiller->isWaiting()) {
+        fulfiller->reject(kj::cp(ex));
+      }
+      // Cause the stream's read() to throw...
+      kj::throwRecoverableException(kj::mv(ex));
```

Comment:
It looks like throwing from here doesn't actually do anything, because the promise that calls this is simply `attach()`ed to the pipe end and no one ever listens for the result.

Reply:
Yep, once I was through adding the `errorBody` back in and reworking a bunch of this I was going to ask what the best way to handle this particular error is as I'm not really sure about the best way to propagate it. Specifically, given how things are wired up here, what is the best way to propagate the error to reads on the connection stream? I was thinking a kj::Canceler but is there another way?